#!/usr/bin/env python
PACKAGE = "depth_controller"

import roslib
roslib.load_manifest(PACKAGE)
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("controller_type",    int_t,    0, "0=PD, 1=SMC, 2=integral-SMC, 3=PID, 4=SMC + separate i", 0,  0, 4)
gen.add("k_p", double_t, 0, "Parameter PD",    9.0, 0)
gen.add("k_d", double_t, 0, "Parameter  PD",    5.8, 0)
gen.add("alpha", double_t, 0, "Parameter SMC",    1.5, 0)
gen.add("Lambda", double_t, 0, "Parameter SMC",    1.5, 0)
gen.add("kappa", double_t, 0, "Parameter SMC",    2.5, 0)
gen.add("epsilon", double_t, 0, "Parameter SMC",    0.4, 0)
gen.add("int_sat", double_t, 0, "Maximum integrated Error",    0.01, 0)

# gen.add("min_setup_time", double_t, 0, "Minimum Calibration Time",    30.0, 0)
# gen.add("max_setup_time", double_t, 0, "Maximum Calibration Time",    60.0, 0)
# gen.add("max_ss_error", double_t, 0, "Maximum Permitted Steady State Error",    0.005, 0)
# gen.add("k_i", double_t, 0, "Parameter PID for Calibration",    0.2, 0)
# gen.add("simulated_offset", double_t, 0, "Simulate Net Bouyancy",    0.0, -0.5, 0.5)

deep_depth_limit = -0.8
shallow_depth_limit = -0.1
gen.add("trajectory",    int_t,    0, "0=Sine Wave, 1=Step Wave, 2=Ramp Wave, else=const", 0,  0, 7)
gen.add("frequency", double_t, 0, "Frequency of Setpoint Trajectory",    0.15, 0, 0.2)
gen.add("mean", double_t, 0, "Mean of Setpoint Trajectory",    -0.5, deep_depth_limit+0.1, shallow_depth_limit-0.1)
gen.add("amplitude", double_t, 0, "Amplitude of Setpoint Trajectory",    0.2, 0, 0.25)

gen.add("surface_pressure", double_t, 0, "Surface Pressure",    1.01325e5, 0)
gen.add("rho", double_t, 0, "SMO Parameter",    2.5, 0)
gen.add("phi", double_t, 0, "SMO Parameter",    0.3, 0)
gen.add("tau", double_t, 0, "SMO Parameter",    0.1, 0)

gen.add("tune_parameters",   bool_t,   0, "Tune Control Parameters",  True)

exit(gen.generate(PACKAGE, "depth_control", "DepthControl"))
